File: LeetCode/src/main/java/com/jack/twopoints/LeetCode11.py
from typing import List


class Solution:
    def maxArea(self, height: List[int]) -> int:
        left, right = 0, len(height) - 1
        res = 0
        while left < right:
            low = min(height[i], height[j])
            res = max(res, low*(right-left))
            if height[left] < height[right]:
                left += 1
            else:
                right -= 1
        return res


if __name__ == "__main__":
    pass

File: LeetCode/src/main/java/com/jack/twopoints/LeetCode11.java
package com.jack.twopoints;

public class LeetCode11 {
    public static void main(String[] args) {
        
    }

    static class Solution{
        public int maxArea(int[] height) {
            int maxArea = 0;
            int len = height.length;
            int left = 0;
            int right = len-1;

            while(left< right){
                int min = Math.min(height[left], height[right]);
                maxArea = Math.max(maxArea, min*(right-left));
            }
            
            return maxArea;
        }

    }

}
